unit_test/sparse/Test_Sparse_spgemm.hpp

```diff
@@ -309,7 +310,7 @@ void test_spgemm(lno_t numRows, size_type nnz, lno_t bandwidth,
         }
         // if size_type is larger than int, mkl casts it to int.
         // it will fail if casting cause overflow.
-        if (input_mat.values.extent(0) > max_integer) {
+        if (A.values.extent(0) > max_integer) {
```

Comment:
Need to check all extents against max_integer here? If this is protecting a code path in the production code, this check should really be moved to the production code where an exception case raised and then we can try-catch the exception here.

Reply:
I noticed this too - luckily the MKL wrapper also has the same check:
```
//if size_type != int...
      const int max_integer = 2147483647;
      if (entriesB.extent(0) > max_integer ||
          entriesA.extent(0) > max_integer) {
        throw std::runtime_error(
            "MKL requires integer values for size type for SPGEMM. Copying to "
            "integer will cause overflow.\n");
        return;
      }
//then convert rowmaps to int
```